pkg/kyverno/apply/command.go

```diff
@@ -207,7 +207,7 @@ func applyCommandHelper(resourcePaths []string, cluster bool, policyReport bool,
 
 	// empty the previous contents of the file just in case if the file already existed before with some content(so as to perform overwrites)
 	// the truncation of files for the case when mutateLogPath is dir, is handled under pkg/kyverno/apply/common.go
-	if !mutateLogPathIsDir {
+	if !mutateLogPathIsDir && mutateLogPath != "" {
```

Comment:
@NoSkillGirl - can you provide more details about this? I read the issue description but it's not clear what's the issue here.

Reply:
When user want to save the mutated result in a file/folder, path of file/folder (where the mutated results is to be saved) is saved in `mutateLogPath`. Variable `mutateLogPathIsDir` is a boolean which used to save the passed path is a directory or not. Variable `mutateLogPathIsDir` is by default false. 

In current code:
`if !mutateLogPathIsDir {}` - even if user does not pass the file/folder path as `mutateLogPathIsDir` is by default false the logic under the if condition will run, which gives error while opening the file (as file path is nil).

Code change:
`if !mutateLogPathIsDir && mutateLogPath != "" {}` - with this condition, if user does not pass any mutated file/folder path, the logic will be skipped.